huaweicloud/resource_huaweicloud_vpc_subnet_v1.go

```diff
@@ -208,12 +208,16 @@ func resourceVpcSubnetV1Read(d *schema.ResourceData, meta interface{}) error {
 	}
 	resourceTags, err := tags.Get(vpcSubnetV2Client, "subnets", d.Id()).Extract()
 	if err != nil {
-		return fmt.Errorf("Error fetching HuaweiCloud VpcSubnet tags: %s", err)
-	}
-
-	tagmap := tagsToMap(resourceTags.Tags)
-	if err := d.Set("tags", tagmap); err != nil {
-		return fmt.Errorf("Error saving tags for HuaweiCloud VpcSubnet (%s): %s", d.Id(), err)
+		if _, ok := err.(golangsdk.ErrDefault404); ok {
+			log.Printf("[INFO] fetching Subnet %s tags failed: The API does not exist or has not been published", d.Id())
```

Comment:
seems 404 clould also come out if the subnet is not exist or something, maybe just warn we got a 404 here is enough?

Reply:
if the subnet is not exist, an error will raise in subnet.Get(), so we don't have to check more.

Reply:
or Maybe we can extract the real error message from the err object. 404 can be possible be other issues.

Reply:
https://github.com/huaweicloud/golangsdk/blob/master/errors.go#L123 seems we format the err already.

Reply:
ok, I will pull a new patch. thanks